tests/SpecTests/ClientSideEncryptionSpecTest.php

```diff
@@ -0,0 +1,757 @@
+
+        $class = Int64::class;
+
+        $intAsString = sprintf((string) $data->value);
+        $array = sprintf('a:1:{s:7:"integer";s:%d:"%s";}', strlen($intAsString), $intAsString);
+        $int64 = sprintf('C:%d:"%s":%d:{%s}', strlen($class), $class, strlen($array), $array);
+
+        return unserialize($int64);
+    }
+
+    private function createTestCollection($jsonSchema)
+    {
+        $options = empty($jsonSchema) ? [] : ['validator' => ['$jsonSchema' => $jsonSchema]];
+        $operation = new CreateCollection($this->getContext()->databaseName, $this->getContext()->collectionName, $options);
+        $operation->execute($this->getPrimaryServer());
+    }
+
+    private function encryptCorpusValue(stdClass $data, ClientEncryption $clientEncryption)
+    {
+        $encryptionOptions = [
+            'algorithm' => $data->algo === 'rand' ? ClientEncryption::AEAD_AES_256_CBC_HMAC_SHA_512_RANDOM : ClientEncryption::AEAD_AES_256_CBC_HMAC_SHA_512_DETERMINISTIC,
+        ];
+
+        switch ($data->identifier) {
+            case 'id':
+                $keyId = $data->kms === 'local' ? 'LOCALAAAAAAAAAAAAAAAAA==' : 'AWSAAAAAAAAAAAAAAAAAAA==';
+                $encryptionOptions['keyId'] = new Binary(base64_decode($keyId), 4);
+                break;
+
+            case 'altname':
+                $encryptionOptions['keyAltName'] = $data->kms === 'local' ? 'local' : 'aws';
+                break;
+
+            default:
+                throw new UnexpectedValueException('Unexpected value "%s" for identifier', $data->identifier);
+        }
+
+        if ($data->allowed) {
+            $encrypted = $clientEncryption->encrypt($this->craftInt64($data), $encryptionOptions);
+            $this->assertEquals($data->value, $clientEncryption->decrypt($encrypted));
+
+            return $encrypted;
+        }
+
+        try {
+            $clientEncryption->encrypt($data->value, $encryptionOptions);
+            $this->fail('Expected exception to be thrown');
+        } catch (RuntimeException $e) {
+        }
+
+        return $data->value;
+    }
+
+    private function insertKeyVaultData(array $keyVaultData = null)
+    {
+        if (empty($keyVaultData)) {
+            return;
+        }
+
+        $context = $this->getContext();
+        $collection = $context->selectCollection('admin', 'datakeys', ['writeConcern' => new WriteConcern('majority')] + $context->defaultWriteOptions);
+        $collection->drop();
+        $collection->insertMany($keyVaultData);
+
+        return;
+    }
+
+    private function prepareCorpusData(stdClass $data, ClientEncryption $clientEncryption)
+    {
+        if ($data->method === 'auto') {
+            $data->value = $this->craftInt64($data);
+
+            return $data;
+        }
+
+        $returnData = clone $data;
+        $returnData->value = $this->encryptCorpusValue($data, $clientEncryption);
+
+        return $data->allowed ? $returnData : $data;
+    }
+
+    private function getAWSCredentials()
```

Comment:
Since the context handles test context (duh) I have no problem moving this to the context class and calling it from there. Good point 👍

Reply:
Moving it to the `Context` class makes sense. Since these tests don't create a context instance, I made the method static as it still provides test context for these prose tests.